File: 0198-house-robber/0198-house-robber.c
int rob(int* nums, int numsSize){
    int withCurr = nums[numsSize-1], woutCurr = 0;
    for(int i=numsSize-2; i>=0; i--){
        int w = nums[i] + woutCurr;
        int wout = withCurr > woutCurr ? withCurr : woutCurr;
        
        withCurr = w;
        woutCurr = wout;
    }
    
    return withCurr > woutCurr ? withCurr : woutCurr;
}

File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        length = len(nums)
        mem = [(0, 0)] * length # (max with, max without)
        mem[length-1] = (nums[length-1], 0)
        
        for i in range(length - 2, -1, -1):
            mem[i] = (nums[i] + mem[i+1][1], max(mem[i+1]))
        
        return max(mem[0])
            